apps/scratch/src/types/scratch-blocks.ts

```diff
@@ -0,0 +1,19 @@
+import { filterNonNull } from "../utilities/filter-non-null";
+import { StudentActionType } from "./scratch-student-activities";
+import { WorkspaceChangeEvent } from "./scratch-workspace";
+import type VM from "scratch-vm";
+
+export interface BlockLifecycleParams {
+  event: WorkspaceChangeEvent;
+  eventAction: StudentActionType;
+  vm: VM;
+  canEditTask: boolean | undefined;
+  blocks: HTMLElement;
+  filterNonNull: typeof filterNonNull;
+  updateSingleBlockConfigButton: (
+    vm: VM,
+    blocks: HTMLElement,
+    opcode: string,
+    canEditTask: boolean | undefined,
+  ) => void;
```

Comment:
This is fine here but I am still wondering what the thought was putting this into a separate file in a `types` directory rather than directly with where the function is defined that uses this as input.

Personally, I would have tended to put it there so that whenever the function changes I only need to touch that file but maybe I am missing something and there is a good reason for doing so :)